treeherder/webapp/api/intermittents_view.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from django.db.models import Count
+from django.db.models.functions import TruncDate
+from rest_framework import generics
+
+from pagination import CustomPagePagination
+from treeherder.config.settings import OPTION_COLLECTION_HASH_MAP
+from treeherder.etl.common import (get_end_of_day,
+                                   get_tree)
+from treeherder.model.models import (BugJobMap,
+                                     Job,
+                                     Push)
+
+from .serializers import (FailureCount,
+                          FailuresByBugSerializer,
+                          FailuresSerializer)
+
+
+class Failures(generics.ListAPIView):
+    """ List of intermittent failures by date range and tree (project name) """
+
+    serializer_class = FailuresSerializer
+    pagination_class = CustomPagePagination
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = get_tree(self.request.query_params.get('tree').encode('utf-8'))
+
+        queryset = BugJobMap.objects.filter(job__repository_id__in=tree, job__push__time__range=(startday, endday),
+                                            job__failure_classification__id=4).select_related('push')\
+            .values('bug_id').annotate(bug_count=Count('job_id')).values('bug_id', 'bug_count').order_by('-bug_count')
+
+        return queryset
+
+
+class FailuresByBug(generics.ListAPIView):
+    """ List of intermittent failure job details by bug, date range and tree (project name) """
+
+    serializer_class = FailuresByBugSerializer
+    pagination_class = CustomPagePagination
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = get_tree(self.request.query_params.get('tree').encode('utf-8'))
+        bug_id = int(self.request.query_params.get('bug'))
+
+        queryset = BugJobMap.objects.select_related('job', 'push').filter(bug_id=bug_id, job__repository_id__in=tree,
+                                                                          job__push__time__range=(startday, endday))\
+            .values('bug_id', 'job_id', 'job__push__time', 'job__repository__name', 'job__option_collection_hash',
+                    'job__signature__job_type_name', 'job__push__revision', 'job__machine_platform__platform')\
+            .order_by('-job__push__time')
+
+        for item in queryset:
+            test_type = item['job__signature__job_type_name']
+            item['test_suite'] = re.sub(r'.+/', '', test_type)
+
+            hash_lookup = OPTION_COLLECTION_HASH_MAP[item['job__option_collection_hash']]
```

Comment:
I think that if the item isn't found in ``OPTION_COLLECTION_HASH_MAP`` it'll actually throw a ``KeyError`` here that needs to be caught, rather than just setting ``hash_lookup`` to undefined where you can check it like it JS.

Reply:
That's a fair point, since any update in the db might won't be caught here.